pkg/objstore/s3/s3.go

```diff
@@ -485,6 +485,7 @@ func (b *Bucket) Upload(ctx context.Context, name string, r io.Reader) error {
 			PartSize:             partSize,
 			ServerSideEncryption: sse,
 			UserMetadata:         b.putUserMetadata,
+			NumThreads:           4,
```

Comment:
Any reason to set this to 4? Is there a default if this is not set? Should we add a comment with explanation?

Reply:
Happy to add ocmment. This is to make sure we are not prone to internal updates to that number. we can play with adjusting this as well in furure.